integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DataMapJobs.scala

```diff
@@ -108,6 +127,21 @@ class EmbeddedDataMapJob extends AbstractDataMapJob {
 
   override def execute(dataMapFormat: DistributableDataMapFormat): util.List[ExtendedBlocklet] = {
     val spark = SparkSQLUtil.getSparkSession
+    val queryId = spark.sparkContext.getConf.get("queryId", UUID.randomUUID().toString)
+    dataMapFormat.setQueryId(queryId)
+    val tempPath = dataMapFormat.getCarbonTable.getTablePath + "/" + queryId;
```

Comment:
Extract a method to util to create query result path

Reply:
fixed